src/main/perl/Path.pm

```diff
@@ -544,6 +559,99 @@ sub directory
 }
 
 
+=item _make_link
+
+This method is mainly a wrapper over C<LC::Check::link>
+returning the standard CAF::Path return values. Every option
+supported by C<LC::Check::link> is supported. C<NoActionFlag>
+is handled by C<LC::Check::link>.
+
+This in internal method, not supposed to be called directly.
+Either call C<symlink> or C<hardlink> public methods instead.
+
+=cut
+
+sub _make_link
+{
+    my ($self, $path, $target, %opts) = @_;
+    $path = $self->_untaint_path($path, "symlink") || return;
+
+    $self->_reset_exception_fail('symlink');
+
+    my $status = LC::Check::link($path, $target, %opts);
+
+    if ( defined($status) ) {
+        return ($status ? CHANGED : SUCCESS);
+    } else {
+        my $link_type = $opts{hard} ? "hardlink" : "symlink";
+        $self->fail("Failed to create $link_type $path (target=$target)");
+        return;
+    }
+}
+
+=item hardlink
+
+Create a hardlink C<path> whose target is C<target>.
+
+Returns an error if C<path> already exists and is not a 
+symlink. If it exists and is a symlink, it is updated.
+The target must exist.
+
+Returns SUCCESS on sucess if the symlink already existed
+with the same target, CHANGED if the symlink was created
+or updated, undef otherwise.
+
+This method relies on C<_make_link> method to do the real work,
+after enforcing the option saying that it is a symlink.
+
+=cut
+
+sub hardlink
+{
+    my ($self, $path, $target, %opts) = @_;
+
+    $opts{hard} = 1;
+    my $status = $self->_make_link($path, $target, %opts);
+
+    return $status;
+}
+
+
+=item symlink
+
+Create a symlink C<path> whose target is C<target>.
+
+Returns an error if C<path> already exists and is not a 
+symlink, except if this is a file and option C<force> is
+defined and true. If C<path> exists and is a symlink, it is updated.
+By default, the target is not required to exist. If you want to
+ensure that it exists, define option C<nocheck> to true.
+
+Returns SUCCESS on sucess if the symlink already existed
+with the same target, CHANGED if the symlink was created
+or updated, undef otherwise.
+
+This method relies on C<_make_link> method to do the real work,
+after enforcing the option saying that it is a symlink.
+
+=cut
+
+sub symlink
+{
+    my ($self, $path, $target, %opts) = @_;
+
+    $opts{hard} = 0;
+    # Disabled the target existence check if 'nocheck' option not specified
+    unless ( defined($opts{nocheck}) ) {
```

Comment:
try to avoid `unless`, it's very confusing to read, esp in combination with variable names that suggest an inversed logic.
you also do not check the value of `nocheck`?